activejob/lib/active_job/queue_adapters.rb

```diff
@@ -48,5 +48,32 @@ module QueueAdapters
     autoload :SneakersAdapter
     autoload :SuckerPunchAdapter
     autoload :TestAdapter
+
+    ADAPTER = 'Adapter'.freeze
+
+    class << self
+      def instance(name, *adapter_args, &block)
+        hash = ([name, block] << adapter_args).hash
```

Comment:
I don't think this is a correct use of `.hash` method, you are not guaranteed to get a different `hash` value for different `([name, block] << adapter_args)`. See also discussion in #15458.